src/modules/status-bar/status-bar.ts

```diff
@@ -112,4 +132,33 @@ export class StatusBar {
     this.isEncryptedCommunication = protocol === 'https://';
     this.processEngineRoute = route;
   }
+
+  private _disposeAllSubscriptions(): void {
+    this._subscriptions.forEach((subscription: Subscription) => {
+      subscription.dispose();
+    });
+  }
+
+  private async _updateStatusBar(): Promise<void> {
+    const solutionUri: string = this._router.currentInstruction.queryParams.solutionUri;
+    const noSolutionUriSpecified: boolean = solutionUri === undefined;
+
+    if (noSolutionUriSpecified) {
+      const remoteSolutionUri: string = window.localStorage.getItem('processEngineRoute');
+      this.activeSolutionEntry = this._solutionService.getSolutionEntryForUri(remoteSolutionUri);
+    } else {
+      this.activeSolutionEntry = this._solutionService.getSolutionEntryForUri(solutionUri);
+    }
+
+    const solutionIsSet: boolean = this.activeSolutionEntry !== undefined;
+    if (solutionIsSet) {
+
+      const diagramName: string = this._router.currentInstruction.params.diagramName;
+      const diagramIsSet: boolean = diagramName !== undefined;
+      if (diagramIsSet) {
+
+        this.activeDiagram = await this.activeSolutionEntry.service.loadDiagram(diagramName);
+      }
+    }
```

Comment:
```suggestion
     const diagramName: string = this._router.currentInstruction.params.diagramName;
     const diagramIsSet: boolean = diagramName !== undefined;
     const solutionIsSet: boolean = this.activeSolutionEntry !== undefined;

     if (solutionIsSet && diagramIsSet) {
           this.activeDiagram = await this.activeSolutionEntry
             .service
             .loadDiagram(diagramName);
      }
```

you can-- and sometimes should-- use two conditions in the `if`